engine/server/conn.go

```diff
@@ -0,0 +1,270 @@
+package server
```

Comment:
split this code out from `engine/server/server.go` to keep that file's length under control. Nothing new or changed, just copy-paste.